libfwupdplugin/fu-device.h

```diff
@@ -193,6 +193,35 @@ FuDevice	*fu_device_new				(void);
 #define fu_device_get_flashes_left(d)		fwupd_device_get_flashes_left(FWUPD_DEVICE(d))
 #define fu_device_get_install_duration(d)	fwupd_device_get_install_duration(FWUPD_DEVICE(d))
 
+/**
+ * FuDeviceFlags:
+ * @FU_DEVICE_FLAG_NONE:			No flags set
+ * @FU_DEVICE_FLAG_NO_AUTO_INSTANCE_IDS:	Do not add instance IDs from the device baseclass
+ * @FU_DEVICE_FLAG_ENSURE_SEMVER:		Ensure the version is a valid semantic version, e.g. numbers separated with dots
+ * @FU_DEVICE_FLAG_ONLY_SUPPORTED:		Only devices supported in the metadata will be opened
+ * @FU_DEVICE_FLAG_MD_SET_NAME:			Set the device name from the metadata `name` if available
+ * @FU_DEVICE_FLAG_MD_SET_NAME_CATEGORY:	Set the device name from the metadata `category` if available
+ * @FU_DEVICE_FLAG_MD_SET_VERFMT:		Set the device version format from the metadata if available
+ * @FU_DEVICE_FLAG_MD_SET_ICON:			Set the device icon from the metadata if available
+ *
+ * The device internal flags.
+ **/
+typedef enum {
+	FU_DEVICE_FLAG_NONE			= 0,
+	FU_DEVICE_FLAG_NO_AUTO_INSTANCE_IDS	= (1llu << 19),	/* Since: 1.5.5 */
+	FU_DEVICE_FLAG_ENSURE_SEMVER		= (1llu << 21),	/* Since: 1.5.5 */
+	FU_DEVICE_FLAG_ONLY_SUPPORTED		= (1llu << 23),	/* Since: 1.5.5 */
+	FU_DEVICE_FLAG_MD_SET_NAME		= (1llu << 32),	/* Since: 1.5.5 */
+	FU_DEVICE_FLAG_MD_SET_NAME_CATEGORY	= (1llu << 33),	/* Since: 1.5.5 */
+	FU_DEVICE_FLAG_MD_SET_VERFMT		= (1llu << 34),	/* Since: 1.5.5 */
+	FU_DEVICE_FLAG_MD_SET_ICON		= (1llu << 41),	/* Since: 1.5.5 */
+	/*< private >*/
+	FU_DEVICE_FLAG_UNKNOWN			= G_MAXUINT64,
+} FuDeviceFlags;
```

Comment:
while getting this started, is there a better way to make it more gtk-doc friendly?  I notice that the previous device flags take the comment on each line, which is really not useful.

https://fwupd.github.io/fwupd-fwupd-enums.html#FWUPD-DEVICE-FLAG-NO-GUID-MATCHING:CAPS

Reply:
actually I think these `enum` will be fine - it's just the `#define` that do poorly.